src/nvim/msgpack_rpc/channel.c

```diff
@@ -321,13 +332,16 @@ static void parse_msgpack(Channel *channel)
       }
       arena_mem_free(arena_finish(&p->arena));
     } else if (p->type == kMessageTypeResponse) {
-      ChannelCallFrame *frame = kv_last(channel->rpc.call_stack);
-      if (p->request_id != frame->request_id) {
+      const char *client_type = rpc_client_type(channel);
+      ChannelCallFrame *frame = strequal(client_type, "msgpack-rpc")
```

Comment:
seems weird to do string parsing at this point in the codebase. I think we should use an enum, and parse the string right after the client calls nvim_set_client_info() .

Reply:
It is unlikely, but is it OK if some existing clients use an unknown type, which would break backward compatibility?

Reply:
the enum can have a `unknown` item

Reply:
OK. I'll make it.